core/http-auth-aws/src/test/java/software/amazon/awssdk/http/auth/aws/RegionSetTest.java

```diff
@@ -24,32 +29,100 @@
+        tc(" a,b  ,c ,d ,e ,f ,  g ", "a,b,c,d,e,f,g", Arrays.asList("a", "b", "c", "d", "e", "f", "g"))
+    );
+
+    private static final List<String> stringFailList = Arrays.asList(
+        null,
+        "",
+        " ",
+        ", ,",
+        ",,,"
+    );
+
+    private static final List<Case<Collection<String>>> collectionCaseList = Arrays.asList(
+        tc(Arrays.asList("*"), "*", Arrays.asList("*")),
+        tc(Arrays.asList("us-west-2"), "us-west-2", Arrays.asList("us-west-2")),
+        tc(Arrays.asList("us-east-1", "us-west-2"), "us-east-1,us-west-2", Arrays.asList("us-east-1", "us-west-2")),
+        tc(Arrays.asList(" us-west-2 "), "us-west-2", Arrays.asList("us-west-2")),
+        tc(Arrays.asList(" us-east-1", " us-west-2  "), "us-east-1,us-west-2", Arrays.asList("us-east-1", "us-west-2")),
+        tc(Arrays.asList(" a", "b  ", "c ", "d ", "e ", "f ", "  g "),
+           "a,b,c,d,e,f,g",
+           Arrays.asList("a", "b", "c", "d", "e", "f", "g")
+        )
     );
 
-    private static final List<Collection<String>> collectionList = Arrays.asList(
-        Arrays.asList("us-west-1"), Arrays.asList("us-west-1", "us-west-2"), Arrays.asList("us-west-2", "us-west-2"),
-        Arrays.asList("*"), Arrays.asList("us-west-2", "us-west-2", "*")
+    private static final List<Collection<String>> collectionFailList = Arrays.asList(
+        null,
+        Arrays.asList(),
+        Arrays.asList(""),
+        Arrays.asList(" ", ""),
+        Arrays.asList("", "", "")
     );
 
-    private static List<String> stringInputs() {
-        return stringList;
+    private static List<Case<String>> stringCases() {
+        return stringCaseList;
     }
 
-    private static List<Collection<String>> collectionInputs() {
-        return collectionList;
+    private static List<String> stringFailures() {
+        return stringFailList;
+    }
+
+    private static List<Case<Collection<String>>> collectionCases() {
+        return collectionCaseList;
+    }
+
+    private static List<Collection<String>> collectionFailures() {
+        return collectionFailList;
     }
 
     @ParameterizedTest
-    @MethodSource("stringInputs")
-    public void create_withStringInput_succeeds(String input) {
-        RegionSet.create(input);
+    @MethodSource("stringCases")
+    public void create_withStringInput_succeeds(Case<String> stringCase) {
+        RegionSet regionSet = RegionSet.create(stringCase.input);
+        assertEquals(stringCase.asString, regionSet.asString());
+        assertIterableEquals(stringCase.asSet, regionSet.asSet());
     }
 
     @ParameterizedTest
-    @MethodSource("collectionInputs")
-    public void create_withCollectionInput_succeeds(Collection<String> input) {
-        RegionSet.create(input);
+    @MethodSource("stringFailures")
+    public void create_withInvalidStringInput_throws(String input) {
+        assertThrows(Exception.class, () -> RegionSet.create(input));
```

Comment:
if this is Exception.class to account for NPE v/s IllegalArgument, might be better to do the `null` case separately than the other cases, so the specific exception can be asserted. at the minimum, make it RuntimeException.class.